test/rust_analyzer/BUILD.bazel

```diff
@@ -0,0 +1,49 @@
+load("//rust:defs.bzl", "rust_analyzer", "rust_library", "rust_proc_macro", "rust_test")
+
+rust_library(
+    name = "mylib",
+    srcs = ["mylib.rs"],
+    proc_macro_deps = [":proc_macro_dep"],
+    deps = [":lib_dep"],
+)
+
+rust_library(
+    name = "lib_dep",
+    srcs = ["lib_dep.rs"],
+)
+
+rust_proc_macro(
+    name = "proc_macro_dep",
+    srcs = ["proc_macro_dep.rs"],
+)
+
+rust_test(
+    name = "mylib_test",
+    crate = ":mylib",
+    proc_macro_deps = [":extra_proc_macro_dep"],
+    deps = [":extra_test_dep"],
+)
+
+rust_library(
+    name = "extra_test_dep",
+    srcs = ["extra_test_dep.rs"],
+)
+
+rust_proc_macro(
+    name = "extra_proc_macro_dep",
+    srcs = ["extra_proc_macro_dep.rs"],
+)
+
+rust_analyzer(
+    name = "rust_analyzer",
+    testonly = True,
+    targets = [":mylib_test"],
+)
+
+rust_test(
+    name = "rust_project_json_test",
+    srcs = ["rust_project_json_test.rs"],
+    args = ["$(location :rust-project.json)"],
+    data = [":rust-project.json"],
+    harness = False,
```

Comment:
Is there a reason why this couldn't use `harness = True`?

Reply:
Answered in a top level comment.